### PR TITLE
pause: bump to newest image v3.4.1

### LINKED ISSUE #10665: Pause: bump to newest image v3.4.1
reasons:

- 3 weeks ago pause image was promoted to v3.4.1 (https://github.com/kubernetes/kubernetes/pull/98205)
```
❯ docker pull k8s.gcr.io/pause:3.4.1
3.4.1: Pulling from pause
fac425775c9d: Pull complete
Digest: sha256:6c3835cab3980f11b83277305d0d736051c32b17606f5ec59f1dda67c9ba3810
Status: Downloaded newer image for k8s.gcr.io/pause:3.4.1
k8s.gcr.io/pause:3.4.1
```

- `pause:latest` is not latest - last time updated at ***Jan 11, 2017*** (ref: https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/pause)
note: latest pause version available on `gcr.io/google_containers` is 3.2 (and that repo should be replaced with `k8s.gcr.io`):
```
❯ docker pull gcr.io/google_containers/pause:3.2
3.2: Pulling from google_containers/pause
c74f8866df09: Pull complete
Digest: sha256:927d98197ec1141a368550822d18fa1c60bdae27b78b0c004f705f548c07814f
Status: Downloaded newer image for gcr.io/google_containers/pause:3.2
gcr.io/google_containers/pause:3.2
```

- ~minikube tests, more often than not, fail with a message `FATA[0000] no such image "k8s.gcr.io/pause:latest", eg:~ - my bad: this is actually a 'delete' test and does not fail
```
    functional_test.go:730: (dbg) Non-zero exit: out/minikube -p functional-20210301023641-94177 ssh sudo crictl inspecti k8s.gcr.io/pause:latest: exit status 1 (272.439218ms)
        
        -- stdout --
                FATA[0000] no such image "k8s.gcr.io/pause:latest" present 
        
        -- /stdout --
        ** stderr ** 
                ssh: Process exited with status 1
        
        ** /stderr **
```

- changelog (ref: https://github.com/kubernetes/kubernetes/blob/master/build/pause/CHANGELOG.md):
```
3.4.1
    Support for Windows container images (OS Versions: 20H2) was added.(#97322, @claudiubelu)

3.4
    Support for Windows container images (OS Versions: 1809, 1903, 1909, 2004) was added. (#91452, @claudiubelu)

3.3
    update debian-base version to v2.1.0 (#90665, [@justaugustus]

3.2
    The pause container is built with the correct "Architecture" metadata. (#87954, @BenTheElder)

3.1
    The pause container gains a signal handler to clean up orphaned zombie processes. (#36853, @verb)
    pause -v will return build information for the pause binary. (#56762, @verb)

3.0
    The pause container was rewritten entirely in C. (#23009, @uluyol)
```

### PR DESCRIPTION
**Pull request by prezha**
fixes #10665 

as described in the original issue, this pr proposes:
- bumping pause image to the newest version 3.4.1
- ~using a specific version instead of outdated `:latest` tag~ - we still need to test `:latest` tag in general, `pause` is handy being small
- switch from `gcr.io/google_containers` to `k8s.gcr.io` repo
- ~should fix `FATA[0000] no such image "k8s.gcr.io/pause:latest"` error in tests~ - my bad: this is actually a 'delete' test and does not fail
- as for `imageID()` used for `pause-axx64:3.3` specifically in `TestFunctional/serial/CacheCmd/verify_cache_inside_node` test, i've shortened the ids to 12 chars as shown by the `docker images`:
```
❯ docker images |grep pause
REPOSITORY                                      TAG                               IMAGE ID       CREATED         SIZE
k8s.gcr.io/pause-arm64                          3.3                               3d18732f8686   10 months ago   484kB
k8s.gcr.io/pause-amd64                          3.3                               0184c1613d92   10 months ago   683kB
```

changelog (ref: https://github.com/kubernetes/kubernetes/blob/master/build/pause/CHANGELOG.md):
```
3.4.1
    Support for Windows container images (OS Versions: 20H2) was added.(#97322, @claudiubelu)

3.4
    Support for Windows container images (OS Versions: 1809, 1903, 1909, 2004) was added. (#91452, @claudiubelu)

3.3
    update debian-base version to v2.1.0 (#90665, [@justaugustus]

3.2
    The pause container is built with the correct "Architecture" metadata. (#87954, @BenTheElder)

3.1
    The pause container gains a signal handler to clean up orphaned zombie processes. (#36853, @verb)
    pause -v will return build information for the pause binary. (#56762, @verb)

3.0
    The pause container was rewritten entirely in C. (#23009, @uluyol)
```

### CODE REVIEW

File: test/integration/functional_test.go [hunk 1]
```diff
@@ -637,7 +637,7 @@ func validateCacheCmd(ctx context.Context, t *testing.T, profile string) {
 
 	t.Run("cache", func(t *testing.T) {
 		t.Run("add_remote", func(t *testing.T) {
-			for _, img := range []string{"k8s.gcr.io/pause:3.1", "k8s.gcr.io/pause:3.3", "k8s.gcr.io/pause:latest"} {
+			for _, img := range []string{"k8s.gcr.io/pause:3.1", "k8s.gcr.io/pause:3.3", "k8s.gcr.io/pause:3.4.1"} {
```
Comment (medyagh) [inline test/integration/functional_test.go:640]: testing latest was one of the test cases, lets keep that, minikbue cache add should still be tested against :latest

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -1398,7 +1397,7 @@ jobs:
           chmod a+x e2e-*
           chmod a+x minikube-*
           START_TIME=$(date -u +%s)
-          KUBECONFIG=$(pwd)/testhome/kubeconfig MINIKUBE_HOME=$(pwd)/testhome ./e2e-linux-amd64 -minikube-start-args=--driver=docker  -test.run "(TestDockerFlags|TestPreload)" -test.timeout=30m -test.v -timeout-multiplier=1.2 -binary=./minikube-linux-amd64 2>&1 | tee ./report/testout.txt
+          KUBECONFIG=$(pwd)/testhome/kubeconfig MINIKUBE_HOME=$(pwd)/testhome ./e2e-linux-amd64 -minikube-start-args=--driver=docker  -test.run "(TestPause|TestDockerFlags|TestPreload)" -test.timeout=30m -test.v -timeout-multiplier=1.2 -binary=./minikube-linux-amd64 2>&1 | tee ./report/testout.txt
```
Comment (medyagh) [inline .github/workflows/master.yml:1400]: currently I am not sure if we should re-add this test again if it causes flakes on free GH machines

**Review by medyagh — CHANGES_REQUESTED** [review]

File: .github/workflows/master.yml (hunk 2 above)
Comment (prezha) [reply .github/workflows/master.yml]: just reverted this change and then saw your comment :)

**Review by medyagh — APPROVED** [review]

**Review by medyagh — APPROVED** [review]

**Pull request merged**